### PR TITLE
Check journal after subject silent correction

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Also I don't know why but pytest is hiding (or not doing the usual magic related to adding) assertion values:
```
>       assert subject_link == href, "Incorrect subject linked from article page %s" % url
E AssertionError: Incorrect subject linked from article page https://end2end--journal.elifesciences.org/articles/6359851919330415893v1
```
doesn't tell what `subject_link` and `href` are. Running with or without `--assert=plain` doesn't seem to make a difference.

**Comment by thewilkybarkid** [comment]
> We have a caching problem here, as the API changes but Journal appears to still have the old cached copy. Is it feasible to wait for the cache to expire here (or to expire it in some way, like avoiding looking at the page until the subject is changed) or does it take too long?

Currently the article API sets a 5-minute cache. We could follow the technique we used for parallel Behat instances (https://github.com/elifesciences/journal/blob/6b6d07fea2447871e97e76af63ac23d27a80bc5b/web/app_test.php#L10-L18). But, there the cache is on the filesystem, in `end2end` it's Redis so we'd need to make sure each instance is a separate DB or it's included in the cache key.

Or, you could clear the whole cache (`bin/console cache:pool:clear cache.guzzle`). This might conflict with other parallel tests of course (if they're relying on cached data). Don't think it's easy to find the individual entry (entries?).

**Comment by giorgiosironi** [comment]
May be departing too much from the actual production configuration (a different entry point or explicit invalidation), the behavior there is "just wait and it will appear". Could we lower the caching ttl in `lax--end2end` to speed up end2end but test the same behavior (expiration rather than explicit invalidation)?

**Comment by thewilkybarkid** [comment]
That would work, as long as it's cacheable it would behave the same (even seconds rather than minutes). How long it caches for isn't important. Might be useful to do that everywhere too.

**Comment by giorgiosironi** [comment]
Depends on https://github.com/elifesciences/builder-private/pull/158

**Comment by giorgiosironi** [comment]
Weird, I can see a good response from
https://end2end--gateway.elifesciences.org/articles/5169096986237515893
but 
https://end2end--journal.elifesciences.org/articles/5169096986237515893
seems to remain a 404.

File: spectrum/checks.py [hunk 1]
```diff
@@ -589,16 +588,13 @@ def profile(self, path):
...
         return annotation_links, pager_links
 
+    def _links(self, body, class_name):
+        """Finds out where 0 or more links selected with CSS class_name point to.
 
-    def _link(self, body, class_name):
-        """Finds out where the link selected with CSS class_name points to.
-
-        May return None if there is no actual link with this class on the page"""
+        May return [] if there is no actual link with this class on the page"""
```
[the remark this answers is not in the document]
Comment (giorgiosironi) [reply spectrum/checks.py]: :+1:

**Pull request merged**